hackernews.el

```diff
@@ -389,13 +488,19 @@ their respective URLs."
                    ?s (propertize (format hackernews-score-format score)
                                   'face 'hackernews-score)
                    ?t (hackernews--button-string
-                       'hackernews-link
+                       (if (and hackernews-show-visited-links (member id (button-type-get 'hackernews-link 'hackernews-visited-ids)))
+			   'hackernews-link-visited
+			 'hackernews-link)
                        (format hackernews-title-format title)
-                       (or item-url comments-url))
+                       (or item-url comments-url)
+		       id)
                    ?c (hackernews--button-string
-                       'hackernews-comment-count
+		       (if (and hackernews-show-visited-links (member id (button-type-get 'hackernews-comment-count 'hackernews-visited-ids)))
+			   'hackernews-comment-count-visited
+			 'hackernews-comment-count)
```

Comment:
Instead of hard-coding "normal" and "visited" button types and looking up the ID set twice, why not move this logic to `hackernews--button-string`?

```el
(defun hackernews--button-string (type label url id)
  "Return button string of TYPE pointing to URL with LABEL.
Replace TYPE with the value of its `hackernews-visited-button'
property if `hackernews-show-visited-links' is non-nil and a
button with TYPE and ID is known to have been visited."
  (and hackernews-show-visited-links
       (gethash id (cdr (assq type hackernews--visited-ids)))
       (setq type (button-type-get type 'hackernews-visited-button)))
  (make-text-button label nil 'type type 'help-echo url 'shr-url url 'id id)
  label)

(defun hackernews--render-item (item)
  "..."
  (let* (...)
    (insert
     (format-spec ...
                   ?t (hackernews--button-string
                       'hackernews-link
                       (format hackernews-title-format title)
                       (or item-url comments-url)
                       id)
                   ?c (hackernews--button-string
                       'hackernews-comment-count
                       (format hackernews-comments-format (or descendants 0))
                       comments-url
                       id))))))
```